src/Mvc/test/WebSites/BasicWebSite/Views/RazorComponents/Index.cshtml

```diff
@@ -1,35 +1,8 @@
 ﻿@using BasicWebSite.RazorComponents;
-@model bool;
+
 <h1>Razor components</h1>
-<div id="Greetings">
-    @if (Model)
-    {
-        @(await Html.RenderStaticComponentAsync<Greetings>())
-    }
-    else
-    {
-        @(await Html.RenderComponentAsync<Greetings>())
-    }
-</div>
+<div id="Greetings">@(await Html.RenderComponentAsync<Greetings>())</div>
 
-<div id="FetchData">
-    @if (Model)
-    {
-        @(await Html.RenderStaticComponentAsync<FetchData>(new { StartDate = new DateTime(2019, 01, 15) }))
-    }
-    else
-    {
-        @(await Html.RenderComponentAsync<FetchData>(new { StartDate = new DateTime(2019, 01, 15) }))
-    }
-</div>
+<div id="FetchData">@(await Html.RenderComponentAsync<FetchData>(new { StartDate = new DateTime(2019, 01, 15) }))</div>
```

Comment:
Arguably it's quite strange that, if we allow passing params to a root component during prerendering, we don't also have a corresponding API for passing params to a root component during interactivity, e.g.:

```cs
endpoints.MapBlazorHub<FetchData>(selector: "root", parameters: new { StartDate = new DateTime(2019, 01, 15) });
```

Not saying we should do that right now, only just that it's a likely future direction.

Perhaps it's also weird that the API doesn't look more like the following, to allow multiple root components:

```cs
endpoints.MapBlazorHub(blazor =>
{
    blazor.AddRootComponent<FetchData>(selector: "root", parameters: new { StartDate = new DateTime(2019, 01, 15) });
    blazor.AddRootComponent<Sidebar>(selector: "sidebar");
});
```

We could add this in the future.

Reply:
Agreed, but I don't think that would really help because that data will be static. I think we need to get an understand feedback from users about what they want.